airdrop/getter_test.gno

```diff
@@ -0,0 +1,275 @@
+			actualBalance.ToString(),
+		)
+	}
+
+	// Simulate some claims
+	claimedAmount := uint64(300000)
+	mockToken.balances[airdropAddress] -= claimedAmount
+
+	// Update airdrop JSON
+	airdropJSON = AirdropToJSON(airdrop)
+
+	jsonStr = GetRemainingBalance(airdropJSON)
+
+	updatedNode, err := json.Unmarshal([]byte(jsonStr))
+	if err != nil {
+		t.Fatalf("Failed to unmarshal updated JSON: %v", err)
+	}
+
+	updatedRemainingBalance := updatedNode.MustKey("remaining_balance").MustString()
+	if updatedRemainingBalance != "700000" {
+		t.Errorf("Expected updated remaining balance to be 700000, but got %v", updatedRemainingBalance)
+	}
+
+	updatedActualBalance, err := u256.FromDecimal(updatedRemainingBalance)
+	if err != nil {
+		t.Fatalf("Failed to convert updated remaining balance string to uint256: %v", err)
+	}
+
+	updatedExpectedBalance := u256.NewUint(initialBalance - claimedAmount)
+	if updatedActualBalance.Cmp(updatedExpectedBalance) != 0 {
+		t.Errorf(
+			"Expected updated remaining balance to be %s, but got %s",
+			updatedExpectedBalance.ToString(),
+			updatedActualBalance.ToString(),
+		)
+	}
+}
+
+func TestGetClaimStatus(t *testing.T) {
+	airdrop, _ := _setupAirdropConfig(t)
+
+	tests := []struct {
+		name           string
+		claimID        uint64
+		initiallySet   bool
+		expectedStatus bool
+	}{
+		{
+			name:           "Unclaimed ID",
+			claimID:        1,
+			initiallySet:   false,
+			expectedStatus: false,
+		},
+		{
+			name:           "Claimed ID",
+			claimID:        2,
+			initiallySet:   true,
+			expectedStatus: true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if tt.initiallySet {
+				word, bit := airdrop.claimIDToBitmapIndex(tt.claimID)
+				airdrop.claimedBitmap[word] |= 1 << bit
+			}
+
+			jsonStr := AirdropToJSON(airdrop)
+			res := GetClaimStatus(jsonStr, tt.claimID)
+
+			node, err := json.Unmarshal([]byte(res))
+			if err != nil {
+				t.Fatalf("Failed to unmarshal JSON: %v", err)
+			}
+
+			claimID, err := node.MustKey("claim_id").GetString()
+			if err != nil {
+				t.Fatalf("Failed to get claim_id: %v", err)
+			}
+
+			if claimID != strconv.FormatUint(tt.claimID, 10) {
+				t.Errorf("Expected claimID to be %d, but got %v", tt.claimID, claimID)
+			}
+
+			claimed, err := node.MustKey("claimed").GetBool()
+			if err != nil {
+				t.Fatalf("Failed to get claimed: %v", err)
+			}
+
+			if claimed != tt.expectedStatus {
+				t.Errorf("Expected claimed status to be %v, but got %v", tt.expectedStatus, claimed)
+			}
+		})
+	}
+}
+
+func TestAirdropToJSONAndCreateAirdropFromJSON(t *testing.T) {
+	mockToken := NewMockGRC20("Test Token", "TST", 18)
+	airdropAddress := testutils.TestAddress("airdrop_address")
+	refundTo := testutils.TestAddress("refund_to")
+	config := Config{
+		RefundableTimestamp: 1234567890,
+		RefundTo:            refundTo,
+	}
+	airdrop := NewAirdrop(mockToken, config, airdropAddress)
+
+	// Set some claimed bits
+	airdrop.claimedBitmap[0] = 0b101
```

Comment:
What's meaning of using bitmap here?

Reply:
This line added for testing the airdrop claim status. For example, the following like sets the oth and 2nd bits to 1 in the 64bit group:

```go
airdrop.claimedBitmap[0] = 0b101
```
This indicates that IDs 0 and 2 have been claimed.